### DISCUSSION TITLE
Add client's Redirect URI to admin setting

### DISCUSSION THREAD

**Discussion by maximepablo** [oidc-wp/openid-connect-generic#433] [Ideas]
Hello,

Thank a lot for your plugin!

I use bedrock and i have to edit your plugin because of the url structure of bedrock

Is there a reason that you didn't add $redirect_uri to admin setting?

It could be amazing for exemple if you check "Alternate Redirect URI" on the admin that you can add your own redirect_uri

**Comment by timnolte** [comment]
I use an alternative Composer-based install of WordPress with the plugin that also has a different URL structure and don't have any problems. Can you provide more details please.

**Comment by maximepablo** [comment]
Thank you for your answer.

On openid-connect-generic.php l.147

$redirect_uri = site_url( '/openid-connect-authorize' );

This return https://exemple.com/wp/openid-connect-authorize.

If $redirect_uri could be a parameter it could be awesome. It will allow more flexibility

**Comment by timnolte** [comment]
Can you please provide information such as what your URL structure is, what the plugin is saying the Redirect URI is, and what you expect it to be? The Composer-based setup I use on sites has the same `wp/` URL prefix, though we don't use the Alternative Redirect URI option and just use the normal option with parameters. We have no problems with our setup.

**Comment by maximepablo** [comment]
Thank for your reply.

Unfortunatly our OAuth2 servers do not allow for a client redirect URI to contain a query string. 

It's why we try to use the alternate redirect uri option but it generate an url with the suffix "wp" inside:
https://exemple.com/wp/openid-connect-authorize

and not

https://exemple.com/openid-connect-authorize

If we have the possibility to set this redirect uri in the admin panel, i won't need to hack your plugin.

Maybe there is a reason why you don't do it.

Thank again, your plugin is perfect.

**Comment by timnolte** [comment]
Are you saying that when you use the Alternate Redirect URI, and follow the instructions to flush your permalinks, that the URL doesn't work? Our plugin registers that alternative URI with WordPress so it should work.

**Comment by maximepablo** [comment]
Thank for your answer.

I don't want to waste your time.

We use Bedrock and site_url return url with 'wp' (and i can't realy change that)

https://discourse.roots.io/t/site-url-returning-wp/12270

It's why i just asking, maybe for security reason you don't want to add redirect_uri as an editable parameter

**Comment by timnolte** [comment]
There are security reasons to not make that setting available to completely change. We may consider adding a hook that would allow customizing the path. The bigger issue though is that the Redirect URI is not supposed to be a frontend URL like you are making it. Using the `wp` prefix should work without the need to be editing the plugin.

I really need more details on the errors you received and the issues you are having with using the Alternative Redirect URI that is being generated.

**Reply by maximepablo** [reply]
Hello,

Thank again for your answer.

I solve the problem it was a bad configuration in my nginx server it why the alternative redirect uri doesn't response.

Thank a lot for your plugin and your time
